framework_tvb/tvb/core/services/burst_service.py

```diff
@@ -241,24 +241,34 @@ def prepare_indexes_for_simulation_results(self, operation, result_filenames, bu
         indexes = list()
         self.logger.debug("Preparing indexes for simulation results in operation {}...".format(operation.id))
         for filename in result_filenames:
-            index = h5.index_for_h5_file(filename)()
-            h5_class = h5.REGISTRY.get_h5file_for_index(type(index))
-
-            with h5_class(filename) as ts_h5:
-                index.fill_from_h5(ts_h5)
-                index.fill_from_generic_attributes(ts_h5.load_generic_attributes())
-                index.gid = ts_h5.gid.load().hex
-
-            index.fk_parent_burst = burst.gid
-            index.fk_from_operation = operation.id
-            if operation.fk_operation_group:
-                datatype_group = dao.get_datatypegroup_by_op_group_id(operation.fk_operation_group)
+            try:
+                self.logger.debug("Preparing index for filename: {}".format(filename))
+                index = h5.index_for_h5_file(filename)()
+                h5_class = h5.REGISTRY.get_h5file_for_index(type(index))
+
+                with h5_class(filename) as index_h5:
+                    index.fill_from_h5(index_h5)
+                    index.fill_from_generic_attributes(index_h5.load_generic_attributes())
+                    index.gid = index_h5.gid.load().hex
```

Comment:
why not load the gid in the base `fill_from_h5` ?

Reply:
good point